### PR TITLE
Draft: Initial version of Logseq import

### PR DESCRIPTION
**Pull request by mihalik**
I think this is in an okay state for a v1.   I can only test this with the unit tests and look at the snapshots so I'm not sure that everything works as expected but it looks pretty close to the Roam outputs.

### Here are some limtations. Let me know if you have any concerns with these.

* I have an example of Logseq TODOs but don't do any special handling of them.  They write a log of when it is toggled that they hide in their UI but it is in the markdown.  So TODO import is kinda messy.  But it'll be kinda hard to parse because they support different workflows of TODO/DOING/DONE and NOW/LATER/DONE.  And you would probably have to throw away that log.  So I'm just letting it import as markdown.
* Attachments won't import.  But I don't think there is a mechanism to support this.  I don't have any test examples of this.
* Most of the date formats are supported but `dd-MM-yyyy` because it will conflict with `MM-dd-yyyy`.  If a user is using that formats they'll have to convert their graph before importing or it will import incorrectly.
  * Right now I'm assuming  `MM-dd-yyyy` mostly just because that was higher in the Logseq list.  I could be convinced to assume the other or to throw an exception any time the dash format is encountered.  Possibly the exception is the best option to avoid a bad import.  Is the exception text displayed to the user?
* Logseq also support Org mode as an alternative to markdown.  This will throw an exception if there are any Org mode pages.  You can mix Org mode with markdown so a single org page will break an import.  But I thought this was better than doing a partial import.  There is a test for this.
* You might double-check my handling of pages that don't exist.  If it is a link to a daily page (has a date) then I link to the daily page id even if it does not exist.  If it is a link to a standalone page I use `toNoteId()` and still link to it.  So [[Some Page]] is a link to `../somepage`.  

Todo (all done):
* ~I'd still like to find a way to get original timestamps of the files (waiting on Discord response)~
  * Confirmed on discord that it is not possible to get timestamps in an export.  I'm going to make a feature request to add this but they are in the middle of a database rewrite so it seemed unlikely this will happen for a while.
* ~I need to prefix the ids with `logseq-`~
* ~Adding more test cases~
* ~Support for properties, right now they are dropped, at a minimum they should be output as text on the page~
* ~Consider using the `list` and `listItem` Dom library used in roam importer~
* ~Move the properties handling code to a separate file.~

### CODE REVIEW

File: src/convertors/logseq/logseq-convertor.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    this.graphId = graphId
+    this.linkHost = linkHost
+  }
+
+  accept = {'application/json': ['.json']}
+
+  convert({data}: ConvertOptions): ConvertResponse {
+    const parsed = JSON.parse(data)
+
+    if (parsed.version !== 1) {
```
Comment (maccman) [inline src/convertors/logseq/logseq-convertor.ts:36]: `parsed?.version`

File: src/convertors/logseq/logseq-convertor.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+      throw new LogseqConversionError('Only able to convert Logseq file version 1')
+    }
+
+    const notes = parsed.blocks as LogseqNote[]
```
Comment (maccman) [inline src/convertors/logseq/logseq-convertor.ts:40]: You might want to coerce the whole object further up rather than just the blocks.

**Comment by maccman** [comment]
This looks almost there. Maybe clean up all the things you want to clean up on the weekend. And I'll get it merged on Monday.

**Comment by mihalik** [comment]
Sounds good.  I've cleaned up a few things and think it is in an pretty good state.  But be sure and read the limitations I added in the description.

I think the date thing is the only thing bothering me right now and I'm starting to lean toward the exception.

**Comment by maccman** [comment]
Excellent work - thank you. Will get this deployed soon.

**Comment by maccman** [comment]
Can you review this please: https://github.com/team-reflect/reflect-import/pull/23

**Comment by asuth** [comment]
@mihalik I'm blocked on importing my large graph on the TODO/DONE thing. I don't care about the dates attached (especially since there's no ui for that in logseq anyway). So all that really needs to happen is convert `^(TODO|DONE)` to `[ ]` or `[x]`

To some extent also the file attachments. They are just files in the logseq folder and shouldn't be too hard to import? But I probably could live without them. Thanks

**Comment by maccman** [comment]
@asuth want to stretch your typescript fu and open a PR? :)
